sanic/server.py

```diff
@@ -99,6 +99,9 @@ def __init__(self, *, loop, request_handler, error_handler,
         self._request_handler_task = None
         self._request_stream_task = None
         self._keep_alive = keep_alive
+        self.state = state if state else {}
+        if 'requests_count' not in self.state:
+            self.state['requests_count'] = 0
```

Comment:
I'd like to refactor the `serve` and `serve_multiple` to be a Server Class. Then, we could reference the Server itself inside http_protocol.
Another gotcha of `serve` method is the return value/type. it may change depending on the `run_async`. 
a good pattern may just wrap them up as a Server class.